flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -769,9 +769,10 @@ abstract class CodeGenerator(
       case PLUS | DATETIME_PLUS if isTemporal(resultType) =>
         val left = operands.head
         val right = operands(1)
+        val typeName = call.getType().getSqlTypeName()
```

Comment:
Remove this line and use `resultType` instead? This would also simplify the patterns in `generateTemporalPlusMinus`.